File: Maximum Average Subarray I - Leetcode 643/Maximum Average Subarray I - Leetcode 643.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        int n = nums.length;
        double curSum = 0;

        for (int i = 0; i < k; i++) {
            curSum += nums[i];
        }

        double maxAvg = curSum / k;

        for (int i = k; i < n; i++) {
            curSum += nums[i];
            curSum -= nums[i - k];

            double avg = curSum / k;
            maxAvg = Math.max(maxAvg, avg);
        }

        return maxAvg;
    }
}

File: Maximum Average Subarray I - Leetcode 643/Maximum Average Subarray I - Leetcode 643.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        int n = nums.size();
        double cur_sum = 0;

        for (int i = 0; i < k; ++i) {
            cur_sum += nums[i];
        }

        double max_avg = cur_sum / k;

        for (int i = k; i < n; ++i) {
            cur_sum += nums[i];
            cur_sum -= nums[i - k];

            double avg = cur_sum / k;
            max_avg = max(max_avg, avg);
        }

        return max_avg;
    }
};
